packages/manager-react-components/src/components/visibility/visibility-management.spec.tsx

```diff
@@ -0,0 +1,122 @@
+        />
+      );
+    },
+  };
+});
+
+describe('visibility management dropdown part', () => {
+  it('should display visibility with all available and all shown', () => {
+    render(
+      <VisibilityManagement
+        columnsVisibility={[
+          {
+            id: 'label',
+            label: 'Label',
+            enableHiding: true,
+            isDisabled: false,
+            isVisible: () => true,
+            onChange: () => null,
+          },
+          {
+            id: 'price',
+            label: 'Price',
+            enableHiding: true,
+            isDisabled: false,
+            isVisible: () => true,
+            onChange: () => null,
+          },
+        ]}
+      />,
+    );
+
+    const checkboxElements = screen.queryAllByRole('checkbox');
+    expect(checkboxElements[0]).toHaveProperty('checked', true);
+    expect(checkboxElements[0]).toHaveProperty('disabled', false);
+
+    expect(checkboxElements[1]).toHaveProperty('checked', true);
+    expect(checkboxElements[1]).toHaveProperty('disabled', false);
+  });
+
+  it('should display visibility with one disabled and only this disabled is shown', () => {
+    render(
+      <VisibilityManagement
+        columnsVisibility={[
+          {
+            id: 'label',
+            label: 'Label',
+            enableHiding: false,
+            isDisabled: true,
+            isVisible: () => true,
+            onChange: () => null,
+          },
+          {
+            id: 'price',
+            label: 'Price',
+            enableHiding: true,
+            isDisabled: false,
+            isVisible: () => false,
+            onChange: () => null,
+          },
+        ]}
+      />,
+    );
+
+    const checkboxElements = screen.queryAllByRole('checkbox');
+    expect(checkboxElements[0]).toHaveProperty('checked', true);
+    expect(checkboxElements[0]).toHaveProperty('disabled', true);
+
+    expect(checkboxElements[1]).toHaveProperty('checked', false);
+    expect(checkboxElements[1]).toHaveProperty('disabled', false);
+  });
+
+  it('should display visibility with all available and one shown', () => {
+    render(
+      <VisibilityManagement
+        columnsVisibility={[
+          {
+            id: 'label',
+            label: 'Label',
+            enableHiding: true,
+            isDisabled: false,
+            isVisible: () => true,
+            onChange: () => null,
+          },
+          {
+            id: 'price',
+            label: 'Price',
+            enableHiding: true,
+            isDisabled: false,
+            isVisible: () => false,
+            onChange: () => null,
+          },
+        ]}
+      />,
+    );
+
+    const checkboxElements = screen.queryAllByRole('checkbox');
+    expect(checkboxElements[0]).toHaveProperty('checked', true);
+    expect(checkboxElements[0]).toHaveProperty('disabled', false);
+
+    expect(checkboxElements[1]).toHaveProperty('checked', false);
+    expect(checkboxElements[1]).toHaveProperty('disabled', false);
+  });
```

Comment:
This test-case can be removed as the main functionality to test is display of columns and disable.

Reply:
It's fixed